src/scripts/site.js

```diff
@@ -83,40 +83,38 @@ function initClient() {
     }
 
     function createConnectionManager() {
-        return require(['connectionManagerFactory', 'apphost', 'credentialprovider', 'events', 'userSettings'], function (ConnectionManager, appHost, credentialProvider, events, userSettings) {
+        return require(['connectionManagerFactory', 'apphost', 'credentialprovider', 'events', 'userSettings', 'apiclient', 'clientUtils'], function (ConnectionManager, appHost, credentialProvider, events, userSettings, apiClientFactory, clientUtils) {
             appHost = appHost.default || appHost;
 
             var credentialProviderInstance = new credentialProvider();
             var promises = [appHost.init()];
 
-            return Promise.all(promises).then(function (responses) {
+            return Promise.all(promises).then(responses => {
                 var capabilities = Dashboard.capabilities(appHost);
 
                 window.connectionManager = new ConnectionManager(credentialProviderInstance, appHost.appName(), appHost.appVersion(), appHost.deviceName(), appHost.deviceId(), capabilities);
 
                 bindConnectionManagerEvents(window.connectionManager, events, userSettings);
+                clientUtils.serverAddress().then(server => {
+                    if (!server) {
+                        Dashboard.navigate('selectserver.html');
+                        return;
+                    }
 
-                if (!AppInfo.isNativeApp) {
-                    console.debug('loading ApiClient singleton');
-
-                    return require(['apiclient', 'clientUtils'], function (apiClientFactory, clientUtils) {
-                        console.debug('creating ApiClient singleton');
-
-                        var apiClient = new apiClientFactory(Dashboard.serverAddress(), appHost.appName(), appHost.appVersion(), appHost.deviceName(), appHost.deviceId());
-
-                        apiClient.enableAutomaticNetworking = false;
-                        apiClient.manualAddressOnly = true;
-
-                        window.connectionManager.addApiClient(apiClient);
+                    console.debug('creating apiclient singleton');
+                    let parts = server.split('/');
+                    let url = parts[0] + '//' + parts[2];
```

Comment:
How about BaseUrl?

Reply:
Hmmm good question, I think I'll need to figure something out for that. I think we can safely assume that someone would include the base URL in their original page load, which helps a lot.